include/rocksdb/db.h

```diff
@@ -353,21 +353,45 @@ class DB {
   // the i'th returned status will have Status::ok() true, and (*values)[i]
   // will store the value associated with keys[i].
   //
-  // (*values) will always be resized to be the same size as (keys).
+  // (*values) is required to be the same size as (keys).
   // Similarly, the number of returned statuses will be the number of keys.
   // Note: keys will not be "de-duplicated". Duplicate keys will return
   // duplicate values in order.
   virtual std::vector<Status> MultiGet(
       const ReadOptions& options,
       const std::vector<ColumnFamilyHandle*>& column_family,
-      const std::vector<Slice>& keys, std::vector<std::string>* values) = 0;
+      const std::vector<Slice>& keys, std::vector<std::string>* values) {
+    assert(values != nullptr);
+    std::vector<PinnableSlice> pinnable_vector(values->size());
+    for (size_t i = 0; i < values->size(); ++i) {
+      assert(!pinnable_vector[i].IsPinned());
+      pinnable_vector[i].ResetSelf(&(*values)[i]);
+    }
+    std::vector<Status> statuses(keys.size());
+    MultiGet(options, column_family, keys, &pinnable_vector, statuses);
+    assert(statuses.size() == pinnable_vector.size());
+    for (size_t i = 0; i < statuses.size(); ++i) {
+      if (statuses[i].ok() && pinnable_vector[i].IsPinned()) {
+        (*values)[i].assign(pinnable_vector[i].data(),
+                            pinnable_vector[i].size());
+      }
+    }
+    pinnable_vector.clear();
+    return statuses;
+  }
   virtual std::vector<Status> MultiGet(const ReadOptions& options,
                                        const std::vector<Slice>& keys,
                                        std::vector<std::string>* values) {
     return MultiGet(options, std::vector<ColumnFamilyHandle*>(
                                  keys.size(), DefaultColumnFamily()),
                     keys, values);
   }
+  // TODO(Zhongyi): make values reference not pointer?
+  virtual void MultiGet(const ReadOptions& options,
```

Comment:
Should we add inline comments about the requirements for values and statuses to be sized properly?